<a href="https://colab.research.google.com/github/d-melamed/DS-Unit-1-Sprint-2-Statistics/blob/master/module3/DM_Copy_of_LS_DS_123_Confidence_Intervals_Assignment_No_Answers.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Confidence Intervals

Soft drinks like Coke and Pepsi are manufactured to have a standard caffeine content. For example, a 12-oz serving of Coke has 34mg of caffeine, and a 12-oz serving of Pepsi has 37.6mg of caffeine. However, fountain soft drinks are typically mixed in individual restaurant dispensers, so it is more difficult to maintain a standard level of caffeine per serving. In this study, researchers randomly sampled Coke, Diet Coke, Pepsi, and Diet Pepsi at a set of franchise restaurants and measured the caffeine content in 12oz of each soft drink2. The data is found in the Soda.xlsx dataset.

Because individuals can be sensitive to caffeine – and because the manufacturers are interested in product consistency – we wish to estimate the mean caffeine content in 12oz of Coke served in franchise restaurants using a 95% confidence interval. 

You can find the Coke data here: 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv'

The first variable is the sample ID and the second variable is the caffeine content in the 12-oz sample measured in mg.

Source: A.N. Garand and L.N. Bell (1997). "Caffeine Content of Fountain and Private-Label Store Brand Carbonated Beverages," Journal of the American Dietetic Association, Vol. 97, #2, pp. 179-182.




###1) Load the dataset and print the first few rows

In [11]:
#Import statements
!wget 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv'


--2020-09-09 20:42:28--  https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.0.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 492 [text/plain]
Saving to: ‘Coke.csv.1’

Coke.csv.1          100%[===================>]     492  --.-KB/s    in 0s      

2020-09-09 20:42:28 (7.03 MB/s) - ‘Coke.csv.1’ saved [492/492]



In [12]:
#Print the top 5 rows
import pandas as pd
import numpy as np
df = pd.read_csv('/content/Coke.csv')

df.head()

Drink  Caffeine
0      1     47.32
1      2     43.78
2      3     48.12
3      4     43.25
4      5     46.42

In [13]:
df.isnull().sum()

Drink       0
Caffeine    0
dtype: int64

###2) Calculate the mean, SD, SE caffeiene content and n for the sample.  Summarize your results in a sentence or two.

In [14]:
#Calculations here

mean = df['Caffeine'].mean()

SD = df['Caffeine'].std()

n = df['Caffeine'].count()

SE = SD/np.sqrt(n)

print(mean, SD, n, SE)

37.9402 5.243756828216712 50 0.7415792024250598


Sentence or two summary.

***The mean caffeine content from this sample is 37.94mg. 95% of coke can caffeine content will fall +/- of around 10 mg from the mean. 50 observations were included in this sample.***

###3) Find t* for a 95% confidence interval.  

Use the starter code below and fill in the degrees of freedom.

In [15]:
from scipy.stats import t

#Don't worry too much about where the 0.975 comes from.  It has to do
#with wanting to determine the *middle* 95% of the t-distribution
#We're going to learn
#how to calculate a 95% CI this easy way in just a minute.

#Recall that n = 223 for the body temp problem.
t_star = t.ppf(0.975,df=n-1)
print('t_star =', t_star)

t_star = 2.009575234489209


###4) Calculate the margin of error for a 95% confidence interval for the mean caffeine content in a 12-oz Coke.



In [16]:
#Calculations here
MoE = t_star * SE
MoE


1.49025919960566

Summary here.

***We are 95% confident that the mean caffeine content of any coke can will fall +/- 1.49mg from the mean of 37.94mg***

###5) Calculate a 95% CI for the mean caffeine content in a 12-oz fountain Coke with the CI formula using the summary statistics and t* that you calculated above.

In [23]:
#Calculations here

lower = mean - MoE
upper = mean + MoE

print(lower, ",", upper)

36.44994080039434 , 39.43045919960566


###6) Calculate a 95% CI for the mean caffeiene content in a 12-oz fountain Coke using the t-interval function in Python.

In [24]:
#Calculations here

t.interval(alpha=0.95, df=n-1, loc=mean, scale=SE)

(36.44994080039434, 39.43045919960566)

###7) Compare the two confidence intervals you calculated.  Do they match?  Should they?

***They match exactly, as they should, since the first way is just the manual/underlying work that the t.interval function is using.***

###8) Interpret the meaning of the 95% confidence interval for the mean caffeiene content in a 12-oz fountain Coke. in a sentence or two.

*** We are 95% confident that the mean caffeince content of any Coke sample will be within that range (36.45 - 39.43)***

###9) Using the t-interval Python function, calculate a 90% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [25]:
#Calculations here
t.interval(alpha=0.90, df=n-1, loc=mean, scale=SE)

(36.696904726749196, 39.1834952732508)

***This interval is more precise, i.e., the range is smaller than the 95% CI. Since we are less confident, we can afford to be more precise.***

###10) Using the t-interval Python function, calculate a 99% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [26]:
#Calculations here
t.interval(alpha=0.99, df=n-1, loc=mean, scale=SE)

(35.95280335285685, 39.92759664714315)

***This is a wider range, however it is more accurate.***

## Stretch goals:

###1) The correspondence between confidence intervals and hypothesis tests.

Read [this](https://blog.minitab.com/blog/adventures-in-statistics-2/understanding-hypothesis-tests-confidence-intervals-and-confidence-levels#:~:text=If%20a%20hypothesis%20test%20produces,corresponding%20confidence%20level%20is%2095%25.&text=If%20the%20confidence%20interval%20does,the%20results%20are%20statistically%20significant.) article about the correspondence between confidence intervals and hypothesis tests.  Feel free to read the whole article, but the relevant part can be found under the heading Why P Values and Confidence Intervals Always Agree About Statistical Significance.

Imagine you work for quality control at Coke and are tasked with making sure that the caffeiene content in the fountain beverages served in restaurants is the same as in a 12-oz can of Coke (34mg).  If you believe that the mean caffeiene content in fountain coke is not 34mg, you must re-train the franchise managers to make sure the Coke served has the correct caffeiene level.

Based on the confidence interval you calculated in the assignment, do you believe that the mean caffeiene content is statistically significantly different from 34 mg in a 12-oz serving?


***Based on the confidence interval that was calculated, the mean caffeine content is statistically significantly different from the 34mg, and we can be 99% confident about that. The low end of the CI was 35.95, which is nearly 2mg higher than the 34mg mean.***

###2) If we increased the sample size from 50 to 100 but the sample mean and SD remained the same, describe **two** ways the margin of error would change.  Would the margin of error become smaller or larger?

***The margin of error would be lower on the high end, and higher on the low end. The standard error would be smaller if we increased the sample size, since the two values are inversely proportional (as sample size goes up, SE will go down). So the size of the interval will decrease.***

In [28]:
#Calculations here.

n2 = 100

SE2 = SD/np.sqrt(n2)

Compare MEs here.



In [29]:
MoE2 = t_star * SE2

In [30]:
lower2 = mean - MoE2
upper2 = mean + MoE2

print(lower2, ",", upper2)

36.8864276142332 , 38.993972385766796


In [31]:
print(lower, ",", upper)

36.44994080039434 , 39.43045919960566
